WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -300,6 +307,17 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
         }
     }
 
+    @objc
+    private func pulledToRefresh() {
+        blogDetailsViewController?.pulledToRefresh(with: refreshControl) { [weak self] in
+            guard let self = self else {
+                return
+            }
+
+            self.sitePickerViewController?.blogDetailHeaderView.blog = self.blog
```

Comment:
In the future we'll have to refresh the dashboard too. What do you think of calling the`syncBlogAndAllMetadata(_: completionHandler:)` BlogService method from My Site instead, then apply changes to the site picker vc + blog details vc / blog dashboard vc -- depending on which section is currently selected?

Reply:
I was in doubt about that. I think each childVC should handle their refresh: blog details refresh the blog and metadata while the dashboard refreshes the dashboard.

However, when I pull to refresh I would also expect the blog data to be synced (it doesn't matter if I'm seeing the Dashboard or the site menu). So this kinda opens up to a few complications:

* We'll probably need to use `DispatchGroup` in case of multiple requests (eg.: updating dashboard and thesyncBlogAndAllMetadata — I don't think it's a good experience if we do one request first and then the other)
* We would need not only to call the child VC to execute their request but also to notify them when the request from My Site finished (so they can do something, like blog details updating its tableView)

As for now, I don't think it is worth pursuing the final solution because we don't have the Dashboard request in place, we'll need to revisit this in the future anyway.

What do you think?

Reply:
> As for now, I don't think it is worth pursuing the final solution because we don't have the Dashboard request in place, we'll need to revisit this in the future anyway.

@leandroalonso  That makes sense to me, thanks for the explanation.